src/ImageSharp/Formats/Webp/WebpAnimationDecoder.cs

```diff
@@ -0,0 +1,408 @@
+        /// <param name="frameX">The frame x coordinate.</param>
+        /// <param name="frameY">The frame y coordinate.</param>
+        /// <param name="frameWidth">The width of the frame.</param>
+        /// <param name="frameHeight">The height of the frame.</param>
+        private void AlphaBlend<TPixel>(ImageFrame<TPixel> src, ImageFrame<TPixel> dst, int frameX, int frameY, int frameWidth, int frameHeight)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            Buffer2D<TPixel> srcPixels = src.PixelBuffer;
+            Buffer2D<TPixel> dstPixels = dst.PixelBuffer;
+            Rgba32 srcRgba = default;
+            Rgba32 dstRgba = default;
+            PixelBlender<Rgba32> blender = PixelOperations<Rgba32>.Instance.GetPixelBlender(PixelColorBlendingMode.Normal, PixelAlphaCompositionMode.SrcOver);
+            for (int y = frameY; y < frameY + frameHeight; y++)
+            {
+                Span<TPixel> srcPixelRow = srcPixels.DangerousGetRowSpan(y);
+                Span<TPixel> dstPixelRow = dstPixels.DangerousGetRowSpan(y);
+                for (int x = frameX; x < frameX + frameWidth; x++)
+                {
+                    ref TPixel srcPixel = ref srcPixelRow[x];
+                    ref TPixel dstPixel = ref dstPixelRow[x];
+                    srcPixel.ToRgba32(ref srcRgba);
+                    dstPixel.ToRgba32(ref dstRgba);
+
+                    if (srcRgba.A is not 0)
+                    {
+                        int dstFactorA = dstRgba.A * (255 - srcRgba.A) / 255;
+                        Rgba32 blendResult = blender.Blend(srcRgba, dstRgba, 1.0f / dstFactorA);
+                        dstPixel.FromRgba32(blendResult);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Dispose to background color. Fill the rectangle on the canvas covered by the current frame
+        /// with background color specified in the ANIM chunk.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="imageFrame">The image frame.</param>
+        /// <param name="backgroundColor">Color of the background.</param>
+        private void RestoreToBackground<TPixel>(ImageFrame<TPixel> imageFrame, Color backgroundColor)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            if (!this.restoreArea.HasValue)
+            {
+                return;
+            }
+
+            var interest = Rectangle.Intersect(imageFrame.Bounds(), this.restoreArea.Value);
+            Buffer2DRegion<TPixel> pixelRegion = imageFrame.PixelBuffer.GetRegion(interest);
+            for (int y = 0; y < pixelRegion.Height; y++)
+            {
+                Span<TPixel> pixelRow = pixelRegion.DangerousGetRowSpan(y);
+                for (int x = 0; x < pixelRow.Length; x++)
+                {
+                    ref TPixel pixel = ref pixelRow[x];
+                    pixel.FromRgba32(backgroundColor);
```

Comment:
Can this be done using bulk operations per scanline?

Reply:
I am pretty sure this can be done as bulk from the looks of it, but I could not figure out how so far.

Reply:
I'll have a look

Reply:
Fixed

Reply:
Thanks for your help